### DISCUSSION TITLE
I’m just trying to connect to a wallet not reprogram my phone

### DISCUSSION THREAD

**Discussion by LordSloppy** [WalletConnect/walletconnect-monorepo#2818] [Developer Support] [answered]
Hey if I'm just an normi who uses walletconnect do I need to upgrade anything? I try to follow along with what I got from the website and it was like trying to help me download things to reprogram my phone which is quite concerning and then went to Discord and it sent me here. Thanks.

**Comment by finessevanes** [comment, marked as answer]
@LordSloppy, you shouldn't have to download anything. Just ensure that you have the latest version of the wallet you're using. If you're using MetaMask on iOS, make sure to go to the App Store and manually update the wallet to version 7.0.1.

**Reply by LordSloppy** [reply]
omg thank you soooo much lol

**Reply by LordSloppy** [reply]
![image](https://github.com/WalletConnect/walletconnect-monorepo/assets/137665793/9718913e-22bf-4815-8f17-16e9b86e4cce)

**Reply by LordSloppy** [reply]
So when I do try to connect to Wallet​ connect to now, I keep getting this error through Rainbow wallet, and![image](https://github.com/WalletConnect/walletconnect-monorepo/assets/137665793/0a7f3ad4-cc92-4bcd-837a-65b5e99e302f)
